server/src/test/java/org/elasticsearch/index/IndexSettingsOverrideTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index;
+
+import org.apache.logging.log4j.Level;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.MockLog;
+
+import static org.elasticsearch.cluster.metadata.IndexMetadata.SETTING_INDEX_VERSION_CREATED;
+import static org.elasticsearch.cluster.routing.allocation.ExistingShardsAllocator.EXISTING_SHARDS_ALLOCATOR_SETTING;
+
+@ESTestCase.WithoutSecurityManager
+public class IndexSettingsOverrideTests extends ESTestCase {
+
+    public static IndexMetadata newIndexMeta(String name, Settings indexSettings) {
+        return IndexMetadata.builder(name)
+            .settings(indexSettings(IndexVersion.current(), randomIntBetween(1, 3), randomIntBetween(1, 3)).put(indexSettings))
+            .build();
+    }
+
+    @SuppressForbidden(reason = "manipulates system properties for testing")
+    public void testStatelessMinRefreshIntervalOverride() {
```

Comment:
Instead of a new file, could we put this in IndexSettingsTests?

Reply:
Unfortunately no. Test explicitly manipulates system properties which is forbidden by ES build rules. To bypass this the test uses `@ESTestCase.WithoutSecurityManager` which has type retention only, hence I added new test file in order not to impact all other exiting tests in `IndexSettingsTests` suite

Reply:
Sounds good thanks